routes/userRouter.js

```diff
@@ -1,103 +1,82 @@
-const {BasicStrategy} = require('passport-http');
+const { BasicStrategy } = require('passport-http');
 const express = require('express');
 const jsonParser = require('body-parser').json();
 const passport = require('passport');
 
-const {User} = require('../models/userModel');
+const { User } = require('../models/userModel');
 
 const router = express.Router();
 
 router.use(jsonParser);
 
 
-// NB: at time of writing, passport uses callbacks, not promises
-const basicStrategy = new BasicStrategy((username, password, callback) => {
-  let user;
-  User
-    .findOne({username: username})
-    .exec()
-    .then(_user => {
-      user = _user;
-      if (!user) {
-        return callback(null, false, {message: 'Incorrect username'});
-      }
-      return user.validatePassword(password);
-    })
-    .then(isValid => {
-      if (!isValid) {        
-        return isValid.callback(null, false, {message: 'Incorrect password'});
-      }
-      else {
-        return callback(null, user)
-      }
-    })
-
-
-});
-
-var user_cache = {};
-
-passport.use(basicStrategy);
-
-passport.serializeUser(function(user, next) {
-  let id = user._id;
-  user_cache[id] = user;
-
-  console.log(`USER: ${user}`)
-  next(null, id);
-});
-
-passport.deserializeUser(function(user, next) {
-  next(null, user_cache[id]);
-});
-
-router.use(passport.initialize());
-
```

Comment:
This `basicStrategy` was moved to a more central location. In an app, you'll need it every where without duplication... maybe it can stay where you want it to.